client/client.go

```diff
@@ -304,6 +306,27 @@ func (c *Client) processChannelRequests() {
 			case *agentpb.StartActionRequest_PtSummaryParams:
 				action = actions.NewProcessAction(p.ActionId, c.cfg.Paths.PTSummary, []string{})
 
+			case *agentpb.StartActionRequest_PtMysqlSummaryParams:
+				// Parses the DSN string
+				cfg, err := mysql.ParseDSN(params.PtMysqlSummaryParams.Dsn)
```

Comment:
Can we extract all this logic somewhere and write tests for it.
we also don't support sockets here.

Reply:
Are you asking my to write a function to handle the parsing of this case?

Reply:
Yes, generating flags from input parameters.